File: core/src/main/java/com/nao4j/subtrol/core/service/slice/SubscriptionSliceService.java
package com.nao4j.subtrol.core.service.slice;

import com.nao4j.subtrol.core.document.internal.ExactPeriod;
import com.nao4j.subtrol.core.document.internal.Quantity;
import com.nao4j.subtrol.core.document.internal.Subscription;

import java.util.Collection;
import java.util.Set;

public interface SubscriptionSliceService {

    Collection<ExactPeriod> slice(Subscription subscription, ExactPeriod period);

    Set<Quantity.QuantityType> dimensions();

}

File: src/main/kotlin/com/nao4j/subtrol/service/SubscriptionSliceService.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.model.ExactPeriod
import com.nao4j.subtrol.model.Subscription

interface SubscriptionSliceService {

    fun slice(subscription: Subscription, period: ExactPeriod): Collection<ExactPeriod>

}
